### ISSUE TITLE
Reporters not working

### ISSUE THREAD

**Issue by antonjahn** [docToolchain/asciidoc-linter#3]
Hi @rdmueller, interesting project you started here! This is the repo I found searching for asciidoc linter.

When running on a file with findings, the current version is failing with errors like:
```
~/.../doc/process/build_management$ asciidoc-lint --format console external_dependencies.adoc 
Error processing external_dependencies.adoc: 'Finding' object has no attribute 'line'
```

The issue maybe inconsistent classes/types. The reporters think they get LintReport (aka List LintError), while in fact they get List[Findings]. So either some processing step is skipped, or the reporters should use finding.position.line.

```
class ConsoleReporter(Reporter):
    """Reports findings in console format with colors"""
    
    def format_report(self, report: LintReport) -> str:
        """Format the report with ANSI colors"""
        if not report.errors:
            return "\033[32m✓ No issues found\033[0m"
            
        output = []
        print(report.errors)   # added by me, drops findings
        for error in report.errors:
            location = f"\033[36mline {error.line}\033[0m"   # issue at hand, error isinstance Finding, not LintError
            if hasattr(error, "file"):
                location = f"\033[36m{error.file}:{location}\033[0m"
            
            output.append(f"\033[31m✗\033[0m {location}: {error.message}")
```

**Comment by rdmueller** [comment]
Hey, cool that you found this project!
Did you notice that it was fully coded and set up by GenAI? So let's see how GenAI will fix your issue...

**Comment by rdmueller** [comment]
Do you think the PR could fix the issue?

**Comment by A-n-d-y-K** [comment]
Dear @rdmueller , 
I´m able to reproduce the issue reported by @antonjahn. When applying your fix in [PR #4](https://github.com/docToolchain/asciidoc-linter/pull/4) 

`(.venv) PS C:\Projects\Repo\asciidoc-linter> asciidoc-linter test.adoc
Error processing test.adoc: 'Finding' object has no attribute 'file'
(.venv) PS C:\Projects\Repo\asciidoc-linter> `

content of test.adoc
```
= AsciiDoc Article Title

Some text

== Header 2

asef
```

**Test Environment**
windows 11, Python 3.13.1

**Additional findings**
`setup.py` line 34 is `long_description=open('README.adoc').read(),`
`setup.py` line 34 may `long_description=open('README.adoc', encoding='utf-8').read(),`

`README.adoc` line 90 is `pip install .`
`README.adoc` line 90 may `pip install -e .`

`README.adoc` line 100, 103, 106 is `asciidoc-lint ...`
`README.adoc` line 100, 103, 106 may `asciidoc-linter ...`

**Why I´m here?**
Curiousity  [Folge 242](https://software-architektur.tv/2024/11/29/episode242.html)

**Comment by A-n-d-y-K** [comment]
Dear @rdmueller and @antonjahn,
is there additional information needed or do you want me to create a new issue?

Thanks in advance for your suggestions.

BR

**Comment by rdmueller** [comment]
ok. I guess I have to give it a try myself and see if I can fix this. Hope to find this time this week.

**Comment by toastwaffle** [comment]
I've got a more comprehensive fix in #5; @rdmueller would appreciate your thoughts before I commit more time to updating the tests
